core/modules/replica/protocol.proto

```diff
@@ -465,45 +492,151 @@ message ReplicationResponseEcho {
+    required string def        =  7;    // char*, unsigned int    default value
+    required uint32 length     =  8;    // unsigned int           The length (width) of the column definition
+    required uint32 max_length =  9;    // unsigned int           The maximum length of the column value
+    required uint32 flags      = 10;    // unsigned int           Flags
+    required uint32 decimals   = 11;    // unsigned int           Number of decimals
+    required int32  type       = 12;    // enum enum_field_types  Field type. These are known variants
+                                        //                        of the type:
+                                        //
+                                        //   MYSQL_TYPE_DECIMAL
+                                        //   MYSQL_TYPE_TINY
+                                        //   MYSQL_TYPE_SHORT
+                                        //   MYSQL_TYPE_LONG
+                                        //   MYSQL_TYPE_FLOAT
+                                        //   MYSQL_TYPE_DOUBLE
+                                        //   MYSQL_TYPE_NULL
+                                        //   MYSQL_TYPE_TIMESTAMP
+                                        //   MYSQL_TYPE_LONGLONG
+                                        //   MYSQL_TYPE_INT24
+                                        //   MYSQL_TYPE_DATE
+                                        //   MYSQL_TYPE_TIME
+                                        //   MYSQL_TYPE_DATETIME
+                                        //   MYSQL_TYPE_YEAR
+                                        //   MYSQL_TYPE_NEWDATE
+                                        //   MYSQL_TYPE_VARCHAR
+                                        //   MYSQL_TYPE_BIT
+                                        //   MYSQL_TYPE_TIMESTAMP2
+                                        //   MYSQL_TYPE_DATETIME2
+                                        //   MYSQL_TYPE_TIME2
+                                        //   MYSQL_TYPE_JSON
+                                        //   MYSQL_TYPE_NEWDECIMAL
+                                        //   MYSQL_TYPE_ENUM
+                                        //   MYSQL_TYPE_SET
+                                        //   MYSQL_TYPE_TINY_BLOB
+                                        //   MYSQL_TYPE_MEDIUM_BLOB
+                                        //   MYSQL_TYPE_LONG_BLOB
+                                        //   MYSQL_TYPE_BLOB
+                                        //   MYSQL_TYPE_VAR_STRING
+                                        //   MYSQL_TYPE_STRING
+                                        //   MYSQL_TYPE_GEOMETRY
+}
+
+/// The message representing data of a row from a result set
+message ProtocolResponseSqlRow {
+
+    /// An array of the raw (non-null terminated) byte strings, where each string
+    /// represents a cell.
+    /// Note that the number of elements in the array must match
+    /// the number of fields.
```

Comment:
Is there anything to be said here about encoding of the bytes, is it guaranteed to be utf-8 for example?

Reply:
The type `bytes` which is used here is a variable length array of the 8-bit bytes (similar to `char` in C++), and it's populated from `std::string` (same as `std::basic_string<char>` or `char*` + `length`). This is a raw result set pulled via the MySQL C-level API after executing a query. It's up to a recipient of the Protobuf message how to interpret the byte stream.

Reply:
So what you say is that encoding is not known? I like this new API even less than I did before 🙁

Reply:
One way to detect it is to make the following query:
```
SHOW VARIABLES LIKE '%collation%'
```
Which could return something like:
```
 Variable_name        | Value            
----------------------+-------------------
 collation_connection | latin1_swedish_ci
 collation_database   | latin1_swedish_ci
 collation_server     | latin1_swedish_ci
```
I have also extended the API to return a result obtained for the current connection from:
https://dev.mysql.com/doc/refman/5.7/en/mysql-character-set-name.html
I hope this solves the immediate problem. If that's not enough then the protocol and the API could be expanded to pass (back an forth) the desired character set:
https://dev.mysql.com/doc/refman/5.7/en/mysql-get-character-set-info.html
https://dev.mysql.com/doc/refman/5.7/en/mysql-set-character-set.html
Though, I think this is going to be an overkill for  problem I'm trying to solve in a context of this development. Character sets other than `latin1` are rarely used in the English-speaking world or scientific applications.

I should also mention that nothing in the MySQL connection API implemented in Qserv (no the Replication system) is sampling the encoding. Perhaps, I'm missing something?